packages/atomic/src/components/commerce/atomic-commerce-interface/atomic-commerce-interface.tsx

```diff
@@ -287,14 +287,14 @@ export class AtomicCommerceInterface
       );
```

Comment:
@fbeaudoincoveo @louis-bompart should we even expose an `executeFirstSearch` on the interface? If this component is solution-type agnostic, it would make more sense for it _not_ to expose an `executeFirstSearch`.

Alternatively, if having this method available on the interface is useful, I'd propose we make it trigger a listing refresh or a search call, depending on the solution type.

Reply:
The `execute first request` type of function on atomic should stay on the interface. It can be named differently (I however dont have a nice generic name to propose...)

It should however be adapted to the solution type (ie: call a different Headless function), as you point out.

Reply:
Did this in https://github.com/coveo/ui-kit/pull/4095/commits/40cbd94bdee6340b59c9b672438e831f6ed03546